subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/TestCountLogger.java

```diff
@@ -95,18 +95,15 @@ public void beforeSuite(TestDescriptor suite) {
     @Override
     public void afterSuite(TestDescriptor suite, TestResult result) {
         if (suite.getParent() == null) {
+            rootSuiteResult = result;
             if (failedTests > 0) {
                 logger.error(TextUtil.getPlatformLineSeparator() + summary());
             }
             progressLogger.completed();
-
-            if (result.getResultType() == TestResult.ResultType.FAILURE) {
-                hadFailures = true;
-            }
         }
     }
 
-    public boolean hadFailures() {
-        return hadFailures;
+    public TestResult getRootSuiteResult() {
```

Comment:
⭕️ Should be annotated with `@Nullable`

Reply:
Done in cce52bc6b75e41dad8f426e2dfef712aff147a42.